tests/includes/scheduler/class-test-comment.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * Test Comment scheduler class.
+ *
+ * @package Activitypub\Tests\Scheduler
+ */
+
+namespace Activitypub\Tests\Scheduler;
+
+use Activitypub\Collection\Outbox;
+use Activitypub\Scheduler\Comment;
+
+/**
+ * Test Comment scheduler class.
+ *
+ * @coversDefaultClass \Activitypub\Scheduler\Comment
+ */
+class Test_Comment extends \WP_UnitTestCase {
+	/**
+	 * User ID for testing.
+	 *
+	 * @var int
+	 */
+	protected static $user_id;
+
+	/**
+	 * Post ID for testing.
+	 *
+	 * @var int
+	 */
+	protected static $post_id;
+
+	/**
+	 * Set up test resources.
+	 *
+	 * @param \WP_UnitTest_Factory $factory Factory object.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$user_id = $factory->user->create( array( 'role' => 'author' ) );
+		self::$post_id = $factory->post->create( array( 'post_author' => self::$user_id ) );
+	}
+
+	/**
+	 * Clean up test resources.
+	 */
+	public static function wpTearDownAfterClass() {
+		wp_delete_post( self::$post_id, true );
+		wp_delete_user( self::$user_id );
+	}
+
+	/**
+	 * Test initialization of hooks.
+	 */
+	public function test_init() {
+		$this->assertSame(
+			20,
+			has_action( 'transition_comment_status', array( Comment::class, 'schedule_comment_activity' ) )
+		);
+		$this->assertSame(
+			10,
+			has_action( 'wp_insert_comment', array( Comment::class, 'schedule_comment_activity_on_insert' ) )
+		);
+	}
+
+	/**
+	 * Test scheduling comment activity on approval.
+	 */
+	public function test_schedule_comment_activity_on_approval() {
+		$comment_id    = self::factory()->comment->create(
+			array(
+				'comment_post_ID'  => self::$post_id,
+				'user_id'          => self::$user_id,
+				'comment_approved' => 0,
+			)
+		);
+		$activitpub_id = \Activitypub\Comment::generate_id( $comment_id );
+
+		wp_set_comment_status( $comment_id, 'approve' );
+
+		$post = $this->get_leatest_outbox_item( $activitpub_id );
+		$this->assertSame( $activitpub_id, $post->post_title );
```

Comment:
@pfefferle This test fails because the above is considered [a "local" comment ](https://github.com/Automattic/wordpress-activitypub/blob/update/comment-outbox/includes/transformer/class-factory.php#L87)and [the Transformer called in add_to_outbox](https://github.com/Automattic/wordpress-activitypub/blob/update/comment-outbox/includes/functions.php#L1564) returns an error.

Is there more nuance to what kind of comment we expect an Outbox item for?

Reply:
only comments by registered users with the `activitypub` cap.

Reply:
Adding the cap doesn't seem to fix it. It looks like it expects the comment to already having been federated via ActivityPub or received via ActivityPub.

Reply:
Looks like you changed it in #1186. Adopting that fixes the tests.